plugins/modules/remote_management/redfish/ilo_redfish_config.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2021-2022 Hewlett Packard Enterprise, Inc. All rights reserved.
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: ilo_redfish_config
+short_description: Sets or updates a configuration attribute. For use with HPE iLO operations that require Redfish OEM extensions.
+version_added: 4.1.0
+description:
+    - "Provides an interface to manage configuration attributes."
+options:
+  category:
+    required: true
+    type: str
+    description:
+      - Command category to execute on iLO
+    choices: ['Manager', 'Sessions']
+  command:
+    required: true
+    description:
+      - List of commands to execute on iLO
+    type: list
+    elements: str
+  baseuri:
+    required: true
+    description:
+      - Base URI of iLO
+    type: str
+  username:
+    description:
+      - User for authentication with iLO
+    type: str
+  password:
+    description:
+      - Password for authentication with iLO
+    type: str
+  auth_token:
+    description:
+      - Security token for authentication with OOB controller
+    type: str
+  timeout:
+    description:
+      - Timeout in seconds for URL requests to iLO controller
+    default: 10
+    type: int
+  attribute_name:
+    required: false
+    description:
+      - Name of the attribute
+    type: str
+  attribute_value:
+    required: false
+    description:
+      - Value of the attribute
+    type: str
+requirements:
+    - "python >= 3.8"
+    - "ansible >= 3.2"
+author:
+    - "Bhavya B (@bhavya06)"
+'''
+
+EXAMPLES = '''
+  - name: Disable WINS Registration
+    community.general.ilo_redfish_config:
+      category: Manager
+      command: SetWINSReg
+      baseuri: 15.X.X.X
+      username: Admin
+      password: Testpass123
+      attribute_name: WINSRegistration
+
+  - name: Set Time Zone
+    community.general.ilo_redfish_config:
+      category: Manager
+      command: SetTimeZone
+      baseuri: 15.X.X.X
+      username: Admin
+      password: Testpass123
+      attribute_name: TimeZone
+      attribute_value: Chennai
+'''
+
+CATEGORY_COMMANDS_ALL = {
+    "Manager": ["SetTimeZone", "SetDNSserver", "SetDomainName", "SetNTPServers", "SetWINSReg"],
+    "Sessions": ["GetiLOSessions"]
```

Comment:
To me it looks like `GetiLOSessions` is something that should be part of an _info modue, not a _config module, or its name is very misleading. Which setting does `GetiLOSessions` modify?

Reply:
We have added this functionality to a new module ilo_redfish_info.